### PR TITLE
feat: add r2r loss

### PR DESCRIPTION
**Pull request by bemc22**
In this PR, we implement the Recorrupted-to-Recorrupted (R2R) unsupervised denosing loss [paper](https://openaccess.thecvf.com/content/CVPR2021/html/Pang_Recorrupted-to-Recorrupted_Unsupervised_Deep_Learning_for_Image_Denoising_CVPR_2021_paper.html).

major changes:
* include `r2r.py` in loss module

minor changes:
* include tests for r2r in losses pool `test_loss.py`
* fix date of PR #145 

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/loss/r2r.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self.alpha = alpha
+
+    def forward(self, y, physics, model, **kwargs):
+        r"""
+        Computes the R2R Loss.
+
+        :param torch.Tensor y: Measurements.
+        :param deepinv.physics.Physics physics: Forward operator associated with the measurements.
+        :param torch.nn.Module model: Reconstruction model.
+        :return: (float) R2R loss.
```
Comment (tachella) [inline deepinv/loss/r2r.py:52]: the output is a torch.Tensor right? and not a float

File: deepinv/loss/r2r.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    where :math:`R` is the trainable network, :math:`A` is the forward operator,
+    :math:`y` is the noisy measurement,
+    :math:`z` is the additional Gaussian noise of standard deviation :math:`\eta`,
+    and :math:`\alpha` is a scaling factor.
+
+
+    This loss is statistically equivalent to the supervised loss function defined on noisy/clean image pairs
+    according to authors in https://ieeexplore.ieee.org/document/9577798
+
+    :param float eta: standard deviation of the Gaussian noise used for the perturbation.
```
Comment (tachella) [inline deepinv/loss/r2r.py:34]: maybe add a note

.. note:

    $\eta$ should be chosen equal or close to $\sigma$ to obtain the best performance.

File: deepinv/loss/r2r.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    :param float alpha: scaling factor of the perturbation.
+    """
+
+    def __init__(self, metric=torch.nn.MSELoss(), eta=35, alpha=0.5):
```
Comment (tachella) [inline deepinv/loss/r2r.py:38]: The library uses images in the range [0,1], so I would choose eta in the range [0,1] instead of [0,255]

File: deepinv/loss/r2r.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        super(R2RLoss, self).__init__()
+        self.name = "r2r"
+        self.metric = metric
+        self.eta = eta / 255.0
```
Comment (tachella) [inline deepinv/loss/r2r.py:42]: remove normalization

**Review by tachella — CHANGES_REQUESTED** [review]
Many thanks for the PR!

The R2R loss is looking good (see some minor changes recommended in the comments). 
My main concern is with the large number of modified files with small changes (mostly blank spaces). Can you please remove these changes?
Thanks!

**Comment by tachella** [comment]
@bemc22  sorry actually the issue is with the new version of black!

we are updating black in the main and this should solve the small changes.

File: deepinv/loss/r2r.py (hunk 3 above)
Comment (bemc22) [reply deepinv/loss/r2r.py]: thanks, fixed in [669bf87]

File: deepinv/loss/r2r.py (hunk 4 above)
Comment (bemc22) [reply deepinv/loss/r2r.py]: thanks, fixed in [669bf87]

**Comment by bemc22** [comment]
> @bemc22 sorry actually the issue is with the new version of black!
> 
> we are updating black in the main and this should solve the small changes.

No problem, I am attentive.

File: deepinv/loss/r2r.py (hunk 2 above)
Comment (bemc22) [reply deepinv/loss/r2r.py:38]: thanks, fixed in [58f29cb]

**Review by matthieutrs — APPROVED** [review]
